include/megaapi_impl.h

```diff
@@ -2532,7 +2508,9 @@ class MegaApiImpl : public MegaApp
         vector<string> excludedPaths;
         long long syncLowerSizeLimit;
         long long syncUpperSizeLimit;
-        MegaMutex sdkMutex;
+        std::recursive_timed_mutex sdkMutex;
+        using SdkMutexGuard = std::unique_lock<std::recursive_timed_mutex>;   // (equivalent to typedef)
+        std::atomic<bool> syncPathStateLockTimeout{ false };
```

Comment:
we might run in to some compilation issues with this: I recall some compilator having issues with initializations in the class (though it's c++11 compliant). We might want to initialize this via atomic_init  in ::init

Reply:
Yes, WP does not support C++11 and it was required to adjust those initializations for the latest version. However, it's not expected to maintain that app any more, since UWP is the standard now. So AFAIK it should be OK

Reply:
thanks - let's try it like this for now then, and if it becomes apparent that we need to init manually in the constructor then we can make that change later.   Setting a default initialization in the header is more maintainable if we can do that, eg. a class with multiple constructors can easily end up with only one constructor updated when a new member is added.